controllers/schedule_controller.go

```diff
@@ -68,27 +68,28 @@ func (r *ScheduleReconciler) SetupWithManager(mgr ctrl.Manager) error {
 		Complete(r)
 }
 
-// FetchEffectiveScheduleResource retrieves a list of EffectiveSchedules and filter the one that matches the given schedule.
-// Returns an error if the listing failed, but nil when no matching EffectiveSchedule object was found.
-func (r *ScheduleReconciler) FetchEffectiveScheduleResource(ctx context.Context, schedule *k8upv1alpha1.Schedule) (*k8upv1alpha1.EffectiveSchedule, error) {
+// fetchEffectiveSchedules retrieves a list of EffectiveSchedules and filter the one that matches the given schedule.
+// Returns an error if the listing failed, but empty map when no matching EffectiveSchedule object was found.
+func (r *ScheduleReconciler) fetchEffectiveSchedules(ctx context.Context, schedule *k8upv1alpha1.Schedule) (map[k8upv1alpha1.JobType]k8upv1alpha1.EffectiveSchedule, error) {
 	list := k8upv1alpha1.EffectiveScheduleList{}
 	err := r.Client.List(ctx, &list, client.InNamespace(cfg.Config.OperatorNamespace))
 	if err != nil {
-		return nil, err
+		return map[k8upv1alpha1.JobType]k8upv1alpha1.EffectiveSchedule{}, err
 	}
 	return filterEffectiveSchedule(list, schedule), nil
 }
 
-func filterEffectiveSchedule(list k8upv1alpha1.EffectiveScheduleList, schedule *k8upv1alpha1.Schedule) *k8upv1alpha1.EffectiveSchedule {
-	for _, effectiveSchedule := range list.Items {
-		if effectiveSchedule.GetDeletionTimestamp() != nil {
+func filterEffectiveSchedule(list k8upv1alpha1.EffectiveScheduleList, schedule *k8upv1alpha1.Schedule) map[k8upv1alpha1.JobType]k8upv1alpha1.EffectiveSchedule {
```

Comment:
I don't like about the method name that it does not say what it filters for exactly.
Maybe `onlyReferencesOfSchedule` or alike?

Reply:
Better now?